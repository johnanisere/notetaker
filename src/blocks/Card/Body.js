import styled from "styled-components";

import P from "../../elements/P";

export default styled(P)`
  padding: 4px 12px;
  margin: 0;
  padding-top: 0;
  overflow-y: scroll;
  font-size: 15px;
  font-weight: 500;
  max-width: calc(100% - 15px);
  box-sizing: border-box;
  margin-bottom: 10px;
  height: 84px;
  line-height: 26px;
  color: #888888;
`;
