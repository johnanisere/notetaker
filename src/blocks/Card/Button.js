import styled from "styled-components";

import Button from "../../elements/Button";

export default styled(Button)`
  padding: 0px 12px;
  background-color: #f1f1f1;
  border-radius: 5px;
  box-shadow: 1px 1px 1px 1px #ffffff;
  border: 1px solid rgba(0, 0, 0, 0.05);
  margin-left: 6px;
  font-size: 18px;
  padding: 5px 8px;
  cursor: pointer;
  color: #888888;
`;
