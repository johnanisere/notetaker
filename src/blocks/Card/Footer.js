import styled from "styled-components";

import Footer from "../../elements/Footer";

export default styled(Footer)`
  padding: 11px 12px;
  box-sizing: border-box;
  font-weight: 400;
  font-size: 13px;
  border-top: 1px solid #ffffff;
  box-shadow: 0px -1px 0px 0px rgba(0, 0, 0, 0.1);
  color: #888888;
`;
