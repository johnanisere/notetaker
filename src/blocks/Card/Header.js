import styled from "styled-components";

import H2 from "../../elements/H2";

export default styled(H2)`
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  padding: 12px;
  margin: 0;
  font-size: 24px;
  font-weight: 400;
  padding: 8px 12px;
  button {
    visibility: hidden;
  }
  &:hover {
    button {
      visibility: visible;
    }
  }
`;
