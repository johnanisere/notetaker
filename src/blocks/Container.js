import styled, { css } from "styled-components";

export default styled.div`
  width: 100%;
  display: flex;
  justify-content: flex-start;
  align-items: flex-start;
  flex-direction: column;
  max-width: 1270px;
  ${({ center }) =>
    center &&
    css`
      padding-top: 100px;
      justify-content: center;
      align-items: center;
      max-width: 600px;
    `}
  ${({ wrap }) =>
    wrap &&
    css`
      flex-direction: row;
      flex-wrap: wrap;
      padding-left: 20px;
      padding-top: 30px;
      @media (max-width: 1316px) {
        max-width: 838px;
      }
      @media (max-width: 897px) {
        max-width: 420px;
      }
    `}
`;
