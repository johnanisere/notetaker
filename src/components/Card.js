import React from "react";

import Card from "../blocks/Card";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

export default ({ note, color, date, number, onEdit, onDelete }) => (
  <Card color={color}>
    <Card.Header>
      <Card.Title color={color}>Note {number}</Card.Title>
      <span>
        <Card.Button onClick={onEdit}>
          <FontAwesomeIcon icon="pen" />
        </Card.Button>
        <Card.Button onClick={onDelete}>
          <FontAwesomeIcon icon="trash" />
        </Card.Button>
      </span>
    </Card.Header>
    <Card.Body>{note}</Card.Body>
    <Card.Footer>{date}</Card.Footer>
  </Card>
);
