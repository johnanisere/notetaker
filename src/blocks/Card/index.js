import Body from "./Body";
import Title from "./Title";
import Header from "./Header";
import Button from "./Button";
import styled, { keyframes } from "styled-components";

import Footer from "./Footer";
import Article from "../../elements/Article";

const fadeIn = keyframes`
  from {
    opacity: 0;
  }

  to {
    opacity:1
  }
`;

const Card = styled(Article)`
  border-left: 8px solid ${({ color }) => color || "#ffffff"};
  overflow: hidden;
  max-width: 399px;
  height: 180px;
  margin-right: 20px;
  margin-bottom: 20px;
  box-shadow: 0 1px 8px 0.2px #d7d7e0;
  border-radius: 4px;
  padding: 0;
  box-sizing: border-box;
  background-color: #f2f2f2;
  animation: ${fadeIn} 0.2s linear;
`;

Card.Body = Body;
Card.Title = Title;
Card.Header = Header;
Card.Button = Button;
Card.Footer = Footer;

export default Card;
