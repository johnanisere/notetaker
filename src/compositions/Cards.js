import React from "react";

import Card from "../components/Card";
import Container from "../blocks/Container";

const Cards = ({ notes, onEdit, onDelete }) => (
  <Container wrap="true">
    {notes.map(({ note, color, date, number }, key) => (
      <Card
        key={key}
        note={note}
        date={date}
        color={color}
        number={number}
        onEdit={() => onEdit(key)}
        onDelete={() => onDelete(key)}
      />
    ))}
  </Container>
);

export default Cards;
