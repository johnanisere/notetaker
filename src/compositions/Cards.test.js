import React from "react";
import Adapter from "enzyme-adapter-react-16";
import Enzyme, { shallow } from "enzyme";

import Cards from "./Cards";
import Card from "../components/Card";
import Container from "../blocks/Container";

Enzyme.configure({ adapter: new Adapter() });

it("Cards component exists", () => {
  const wrapper = shallow(<Cards />);
  expect(wrapper.exists()).toBe(true);
});

it("Cards component renders", () => {
  const wrapper = shallow(<Cards />);

  expect(wrapper).toMatchSnapshot();
});

it(`Cards should contain "Container" and "Card"`, () => {
  const mockCallBack = jest.fn();
  const props = {
    notes: [
      {
        color: "blue",
        date: "March 31, 8:54 pm",
        note: "Hey",
        number: 1
      }
    ],
    onEdit: mockCallBack,
    onDelete: mockCallBack
  };
  //   expect.assertions(4);
  const wrapper = shallow(<Cards {...props} />);
  expect(wrapper.find(Card)).toHaveLength(1);
  expect(wrapper.find(Container)).toHaveLength(1);
  //   console.log({ btn: wrapper.props() });

  //   expect(mockCallBack.mock.calls.length).toEqual(1);
  expect(wrapper.find(Container).props().wrap).toBe("true");
});
