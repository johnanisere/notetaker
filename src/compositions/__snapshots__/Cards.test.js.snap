// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards component renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Cards />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [],
      "wrap": "true",
    },
    "ref": null,
    "rendered": Array [],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-gZMcBi",
        "isStatic": false,
        "rules": Array [
          "
  width: 100%;
  display: flex;
  justify-content: flex-start;
  align-items: flex-start;
  flex-direction: column;
  max-width: 1270px;
  ",
          [Function],
          "
  ",
          [Function],
          "
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-gZMcBi",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [],
        "wrap": "true",
      },
      "ref": null,
      "rendered": Array [],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-gZMcBi",
          "isStatic": false,
          "rules": Array [
            "
  width: 100%;
  display: flex;
  justify-content: flex-start;
  align-items: flex-start;
  flex-direction: column;
  max-width: 1270px;
  ",
            [Function],
            "
  ",
            [Function],
            "
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-gZMcBi",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
